pkg/controller/oam/applicationconfiguration/render.go

```diff
@@ -392,8 +390,126 @@ func addDataOutputsToDAG(dag *dependency.DAG, outs []v1alpha2.DataOutput, obj *u
 	}
 }
 
-func addDataInputsToDAG(dag *dependency.DAG, ins []v1alpha2.DataInput, obj *unstructured.Unstructured, attaches []unstructured.Unstructured) {
+func (r *components) handleDependency(ctx context.Context, w *Workload, acc v1alpha2.ApplicationConfigurationComponent, dag *dependency.DAG) (bool, error) {
```

Comment:
Currently there is no plan to support dependency w.r.t. scope. Only workloads and traits.
This is because I do not want to bloat the feature to cover scenarios that do not have real world use cases.